.github/workflows/format_check.yml

```diff
@@ -18,43 +13,37 @@ on:
       - synchronize
     branches:
       - 'main'
-    paths:
-      - '**/*.c'
-      - '**/*.cpp'
-      - '**/*.h'
-      - '**/*.hpp'
-
-  workflow_dispatch:
-    inputs:
-      logLevel:
-        description: 'Log level'     
-        required: true
-        default: 'warning'
 
 jobs:
-  format-check:
+  verify-format:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        path:
-          - check: 'cores/arduino/'
-            exclude: 'cores/arduino/api/'
-          - check: 'loader/'
-            exclude: 'loader/llext_exports\.c$'
-          - check: 'libraries/'
-            exclude: '(examples|extras|ea_malloc)'
-      fail-fast: false
-
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
           submodules: false
           persist-credentials: false
-
+      - name: Get changed source files that need format check
+        id: changed-files
```

Comment:
This is okay, but is running the formatter on all PRs unconditionally that bad? Most of the PRs will likely contain code changes anyway.